### DISCUSSION TITLE
Anyone have success using file router with threejs

### DISCUSSION THREAD

**Discussion by brucewongqbio** [TanStack/router#2095] [Q&A] [answered]
sorry I haven't made a small preview yet but I'm working on a pretty hefty app that involves rendering threejs canvases via react three fiber with tanstack router. For some reason, whenever I put a mesh into my canvas, the entire router stops to render anything. I'll create a minimally reproducible example soon but was wondering if anyone had any situation this has happened where nothing is rendered from the router anymore due to some threejs interaction. Thank you!

**Comment by brucewongqbio** [comment]
So I found out that my problem is that I have some loaders in my threejs component and that tanstack is trying to show a pending component for those loaders. I want to completely disable that feature of tanstack. How can I do that?

**Reply by schiller-manuel** [reply]
did you define a `defaultPendingComponent`? if yes, this is the fallback that is used.

**Reply by brucewongqbio** [reply]
I didn't define a `defaultPendingComponent`. I believe there is some default behavior happening.

**Reply by schiller-manuel** [reply]
Can you please provide a minimal complete example by forking one of the examples on stackblitz?

**Reply by brucewongqbio** [reply]
Is it possible that the pending component is essentially null and just rendering null if nothing is defined? I believe that was the behavior I was seeing.

**Reply by brucewongqbio** [reply]
yes I'll get to that as soon as I can. Currently the suspense catching the loading has been my workaround.

**Reply by schiller-manuel** [reply, marked as answer]
> Is it possible that the pending component is essentially null and just rendering null if nothing is defined? I believe that was the behavior I was seeing.

For the `root` route, this is the case currently.
Try out the packages built for this PR to see if this fixes your issue: https://github.com/TanStack/router/pull/2035/files

**Reply by brucewongqbio** [reply]
Thanks so much! This does look like it will fix it. I'll try it.

**Reply by schiller-manuel** [reply]
let me know if it works so we can merge this PR

**Comment by brucewongqbio** [comment]
I ended up wrapping the component causing the loading with a React.Suspense so that the loading event get's caught there instead of at the router level. Would still like to know how to disable it at the router level if possible. Thank you.
